### PR TITLE
Replace deprecated urlopen cafile parameter

### PR DESCRIPTION
**Pull request by salty-horse**
The `cafile` parameter of `urlopen` has been deprecated since python 3.6, and is [removed in the upcoming 3.13](https://docs.python.org/3.13/whatsnew/3.13.html#urllib).

It's used in the `setup.py recent_uas` command. I replaced it with an SSL context, and cleaned up the json dumping code.

Note: I don't really use this feature - I just searched for deprecated functions.

### CODE REVIEW

File: setup/resources.py [hunk 1]
```diff
@@ -166,8 +166,8 @@ class RecentUAs(Command):  # {{{
     def run(self, opts):
         from setup.browser_data import get_data
         data = get_data()
-        with open(self.UA_PATH, 'wb') as f:
-            f.write(json.dumps(data, indent=2, ensure_ascii=False, sort_keys=True).encode('utf-8'))
+        with open(self.UA_PATH, 'w', encoding='utf-8') as f:
+            json.dump(data, f, indent=2, ensure_ascii=False, sort_keys=True)
```
Comment (eli-schwartz) [inline setup/resources.py:170]: This change is not mentioned in the commit message and is totally unrelated. (It is also just painting the bikeshed...)

Comment (salty-horse) [reply setup/resources.py:170]: I know, and it's atypical of me. I apologize, and will split them up in the future.

It's not *completely* unrelated as it's part of the same "recent UAs" functionality.
